### PR TITLE
PT-414: Allow to pass response groups as additional parameter in CartProductService.GetCartProductsByIdsAsync

### PR DESCRIPTION
**Pull request by akak1977**
https://virtocommerce.atlassian.net/browse/PT-413

### CODE REVIEW

File: src/XPurchase/VirtoCommerce.XPurchase/Services/CartProductService.cs [hunk 1]
```diff
@@ -32,7 +32,7 @@ IItemService productService
             _mapper = mapper;
         }
 
-        public async Task<IEnumerable<CartProduct>> GetCartProductsByIdsAsync(CartAggregate cartAggr, string[] ids)
+        public async Task<IEnumerable<CartProduct>> GetCartProductsByIdsAsync(CartAggregate cartAggr, string[] ids, string additionalResponseGroups = null)
```
Comment (ksavosteev) [inline src/XPurchase/VirtoCommerce.XPurchase/Services/CartProductService.cs:35]: Change interface signature

**Review by ksavosteev — CHANGES_REQUESTED** [review]

**Review by eMazeika — APPROVED** [review]

**Review by ksavosteev — APPROVED** [review]

File: src/XPurchase/VirtoCommerce.XPurchase/Services/CartProductService.cs [hunk 2]
```diff
@@ -41,10 +41,16 @@ public async Task<IEnumerable<CartProduct>> GetCartProductsByIdsAsync(CartAggreg
             if (ids == null)
             {
                 throw new ArgumentNullException(nameof(ids));
+            }
+
+            var itemResponseGroups = ItemResponseGroup.ItemAssets | ItemResponseGroup.ItemInfo | ItemResponseGroup.Outlines | ItemResponseGroup.Seo;
```
Comment (Vectorfield4) [inline src/XPurchase/VirtoCommerce.XPurchase/Services/CartProductService.cs:46]: Should we rename this variable to `defaultResponseGroups`?

File: src/XPurchase/VirtoCommerce.XPurchase/Services/ICartProductService.cs [hunk 3]
```diff
@@ -5,6 +5,6 @@ namespace VirtoCommerce.XPurchase.Services
 {
     public interface ICartProductService
     {
-        Task<IEnumerable<CartProduct>> GetCartProductsByIdsAsync(CartAggregate cartAggr, string[] ids);
+        Task<IEnumerable<CartProduct>> GetCartProductsByIdsAsync(CartAggregate cartAggr, string[] ids, string additionalResponseGroups = null);
```
Comment (Vectorfield4) [inline src/XPurchase/VirtoCommerce.XPurchase/Services/ICartProductService.cs:8]: Should we remove the default parameter?

**Review by Vectorfield4 — CHANGES_REQUESTED** [review]

File: src/XPurchase/VirtoCommerce.XPurchase/Services/CartProductService.cs [hunk 4]
```diff
@@ -41,10 +41,16 @@ public async Task<IEnumerable<CartProduct>> GetCartProductsByIdsAsync(CartAggreg
             if (ids == null)
             {
                 throw new ArgumentNullException(nameof(ids));
+            }
+
+            var defaultResponseGroups = ItemResponseGroup.ItemAssets | ItemResponseGroup.ItemInfo | ItemResponseGroup.Outlines | ItemResponseGroup.Seo;
+            if (!string.IsNullOrEmpty(additionalResponseGroups))
+            {
+                defaultResponseGroups |= EnumUtility.SafeParseFlags(additionalResponseGroups, ItemResponseGroup.ItemLarge);
```
Comment (eMazeika) [inline src/XPurchase/VirtoCommerce.XPurchase/Services/CartProductService.cs:49]: there is no need to fallback to ItemLarge. ItemAssets as default value should be enough

**Review by eMazeika — APPROVED** [review]

**Review by ksavosteev — APPROVED** [review]

**Review by Vectorfield4 — APPROVED** [review]

File: src/XPurchase/VirtoCommerce.XPurchase/Services/ICartProductService.cs (hunk 3 above)
Comment (akak1977) [reply src/XPurchase/VirtoCommerce.XPurchase/Services/ICartProductService.cs:8]: We decided to keep this

**Pull request merged**